### PR TITLE
User following system AB#309 AB#314 AB#316 AB#318 AB#319 AB#320

### PR DESCRIPTION
**Pull request by kikz0rsk**

### CODE REVIEW

File: Controllers/UserController.cs [hunk 1]
```diff
@@ -28,5 +40,58 @@ public async Task<ActionResult<UserResponseDTO>> UpdateUser([FromForm] UserUpdat
 
             return Ok(_mapper.Map<UserResponseDTO>(result));
         }
+
+        [HttpGet("posts")]
```
Comment (Kesuera) [inline Controllers/UserController.cs:44]: remove

File: Controllers/UserController.cs [hunk 2]
```diff
@@ -28,5 +40,58 @@ public async Task<ActionResult<UserResponseDTO>> UpdateUser([FromForm] UserUpdat
...
+        public async Task<ActionResult<IEnumerable<UserPostResponseDTO>>> GetUserPosts(
+            [FromQuery] PaginationRequestDTO request
+        )
+        {
+            var result = await _postService.GetUserPosts(request, GetUserAddress());
+
+            return Ok(_mapper.Map<IEnumerable<UserPostResponseDTO>>(result));
+        }
+
+        [HttpGet("chat-profile")]
```
Comment (Kesuera) [inline Controllers/UserController.cs:54]: remove

File: Controllers/UserController.cs [hunk 3]
```diff
@@ -28,5 +40,58 @@ public async Task<ActionResult<UserResponseDTO>> UpdateUser([FromForm] UserUpdat
...
+        [HttpGet("followings")]
+        public async Task<ActionResult<IEnumerable<FollowerResponseDTO>>> GetUserFollowings(
+            [FromQuery] PaginationRequestDTO request)
+        {
+            var result = await _followerService.GetUserFollowings(GetUserAddress(), request);
+
+            return Ok(_mapper.Map<IEnumerable<FollowerResponseDTO>>(result));
+        }
+
+        [HttpPost("followings")]
```
Comment (Kesuera) [inline Controllers/UserController.cs:81]: prefix URL with userAddress and use BindUser annotation like in other controllers, no need to check whether user exists in code afterwards

FollowRequestDTO can be deleted afterwards

File: Controllers/UserController.cs [hunk 4]
```diff
@@ -28,5 +40,58 @@ public async Task<ActionResult<UserResponseDTO>> UpdateUser([FromForm] UserUpdat
...
+        public async Task<ActionResult<IEnumerable<FollowerResponseDTO>>> FollowUser(FollowRequestDTO request)
+        {
+            var result = await _followerService.FollowUser(GetUserAddress(), request.UserAddress);
+
+            return Ok(_mapper.Map<FollowerResponseDTO>(result));
+        }
+
+        [HttpDelete("followings")]
```
Comment (Kesuera) [inline Controllers/UserController.cs:89]: prefix URL with userAddress and use BindUser annotation like in other controllers, no need to check whether user exists in code afterwards

FollowRequestDTO can be deleted afterwards

File: Controllers/UserController.cs [hunk 5]
```diff
@@ -28,5 +40,58 @@ public async Task<ActionResult<UserResponseDTO>> UpdateUser([FromForm] UserUpdat
...
+
+        [HttpGet("chat-profile")]
+        public async Task<ActionResult<UserChatProfileResponseDTO>> GetUserChatProfile()
+        {
+            var result = await _chatService.GetUserChatProfile(GetUserAddress());
+
+            return Ok(_mapper.Map<UserChatProfileResponseDTO>(result));
+        }
+
+        [HttpGet("followers")]
```
Comment (Kesuera) [inline Controllers/UserController.cs:62]: prefix URL with userAddress and use BindUser annotation, we also need endpoints for getting the followers of other users, not just the currently logged in user

File: Controllers/UserController.cs [hunk 6]
```diff
@@ -28,5 +40,58 @@ public async Task<ActionResult<UserResponseDTO>> UpdateUser([FromForm] UserUpdat
...
+        public async Task<ActionResult<IEnumerable<FollowerResponseDTO>>> GetUserFollowers(
+            [FromQuery] PaginationRequestDTO request)
+        {
+            var result = await _followerService.GetUserFollowers(GetUserAddress(), request);
+            var dto = result.Select(x => new FollowerResponseDTO { UserAddress = x.UserAddress, FollowingSince = x.CreatedAt });
+
+            return Ok(dto);
+        }
+
+        [HttpGet("followings")]
```
Comment (Kesuera) [inline Controllers/UserController.cs:72]: prefix URL with userAddress and use BindUser annotation, we also need endpoints for getting the followings of other users, not just the currently logged in user

File: Models/DTOS/HTTP/Requests/FollowRequestDTO.cs
Comment (Kesuera) [inline Models/DTOS/HTTP/Requests/FollowRequestDTO.cs:1]: delete

File: Models/DTOS/HTTP/Requests/Validators/FollowRequestDTOValidator.cs
Comment (Kesuera) [inline Models/DTOS/HTTP/Requests/Validators/FollowRequestDTOValidator.cs:1]: delete

File: Models/DTOS/HTTP/Responses/FollowerResponseDTO.cs
Comment (Kesuera) [inline Models/DTOS/HTTP/Responses/FollowerResponseDTO.cs:1]: Replace UserAddress with UserResponseDTO

**Review by brano-hozza — APPROVED** [review]

**Pull request merged**